pkg/policy/mapstate.go

```diff
@@ -137,16 +182,16 @@ func (keys MapState) RedirectPreferredInsert(key Key, entry MapStateEntry) {
 // from the localhost. It inserts the Key corresponding to the localhost in
 // the desiredPolicyKeys if the localhost is allowed to communicate with the
 // endpoint.
-func (keys MapState) DetermineAllowLocalhostIngress(l4Policy *L4Policy) {
+func (keys MapState) DetermineAllowLocalhostIngress() {
```

Comment:
Unless I missed something, it will be up to the users to be careful not to block ingress from host with a deny policy. How can they use L4-only deny policies without breaking allow-all-from-host?
```yaml
  ingressDeny:
    toPorts:
    - ports:
      - port: "80"
        protocol: TCP 
```

Reply:
This is an interesting question and came up while I was syncing with Andre out-of-band too.

If the user explicitly writes a policy that says "deny from host", then it's a little bit hard to argue why the flag for automatic allow-all would take precedence. The default value of that flag allows from host, so it is likely not *explicitly* configured by the user. Presumably if the user configures such a policy, Cilium really should drop such traffic.

Are you concerned that users might accidentally write a "drop from host" rule?

Reply:
> Are you concerned that users might accidentally write a "drop from host" rule?

Yes, because any L4-only deny rule is a drop from host rule if I understand correctly.

I think this was addressed during the Cilium community meeting and we'll leave it to the user to be careful not to drop their traffic. Given the denies are explicit in the case of deny policies, it's probably fine.